Src/main.c

```diff
@@ -100,14 +102,32 @@ int main(void)
   debug("\n\nBootcamp starting up...");
   debug("Compiled on %s at %s", __DATE__, __TIME__);
 
-
   HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, GPIO_PIN_SET);
+
+  HAL_ADC_Start(&hadc); //Enables ADC and starts conversion of the regular channels.
+  HAL_TIM_PWM_Start(&htim16, TIM_CHANNEL_1); //starts PWM signal generation. First argument is the pointer to the TIM handle and the second parameter is the channel used or the timer
+
+
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+
+  uint16_t adc_output = 0; //ADC gives a 16 bit output
```

Comment:
Double check the datasheet for bit resolution :)

https://www.st.com/resource/en/datasheet/cd00237391.pdf

Reply:
Note for myself: "3 × 12-bit, 0.5 μs ADCs with up to 24 channels and up to 6 MSPS in triple interleaved mode"

Oh... I'll have to change that then. And if it is 12-bit then the max ADC value is 4095, not 4096. 

Thank you!

Reply:
Alright, I added a bit_field struct in my most recent commit that will make the variable adc_output a 12-bit integer.